Update post

CoBatool opened the pull request:
Closes #10 

- Let user update post
- Show post as updated when someone edits a post
- Update column in DB

**Missing:** 
- Edit attachment



**Edit button**
![1](https://user-images.githubusercontent.com/45091092/98882278-ece43480-2459-11eb-8c31-8245c5b75a3a.png)

**When clicked on edit possibility to edit title/message**
![2](https://user-images.githubusercontent.com/45091092/98882289-efdf2500-2459-11eb-929b-6bf71ba552ad.png)

**Once edited, show timestamp**
![3](https://user-images.githubusercontent.com/45091092/98882292-f1a8e880-2459-11eb-9d08-1d30abf80bf8.png)


**Database updated**
![4](https://user-images.githubusercontent.com/45091092/98882298-f40b4280-2459-11eb-9ea4-14a4d5c82dc2.png)

mjimenez98 reviewed (approved):
This is pretty cool! Love it ❤️ 

One non-blocking thing I noticed is that if I click on edit and I don't change anything, the message will still pop up saying I edited the post. Would it be possible to verify if the original title or message has changed before displaying the edit timestamp?

Also, I would suggest running `Code -> Reformat Code` or `Alt + Command + L` on `PostServlet.java` and `DBPost.java` to get the formatting in line with main

mjimenez98 commented on src/main/webapp/Posts.jsp:
```diff
@@ -1,67 +1,80 @@
+        if (request.getAttribute("posts") != null) {
+            for (Post post : posts) {
+    %>
+    <form action="posts" method="post">
+        <% if (session.getAttribute("editMessage") != null && session.getAttribute("editMessage") == (Integer) post.getId()) {%>
+        <p><input type="text" name="editTitle" value="<%=post.getTitle()%>">  <%=post.getUsername() %></p>
+        <input type="text" name="editMessage" value="<%=post.getMessage()%>">        
+        <input type="submit" name="request" class="btn button-color" value="save">
+        <%} else {%>
+            <p><%= post.getTitle() + " - " + post.getUsername() %>
+            </p>
+            <p><%=post.getMessage()%>
+            </p>
+            <% if (post.getCreatedAt().isBefore(post.getUpdatedAt())){%>
+                <p><i>Edited <%= post.getUpdatedAt()%> </i></p>
+                <%  }
+                }%>
+        <%if (user != null && user.equals(post.getUsername())) { %>
+        <input type="hidden" name="id" value="<%= post.getId() %>">
+        <input type="submit" name="request" class="btn button-color" value="edit">
+        <input type="submit" name="request" class="btn button-color" value="delete">
+    </form>
+    <% } %>
+    <hr/>
+    <%
+            }
+        }
+    %>
```
I am finding it a bit hard to follow where things are being opened and closed, what's in where and so on 😕 . Would it be possible to give it some formatting or different indentation?

CoBatool replied:
I just updated the file with `Ctrl+Alt+L` !   Is this better? Or do you want me to change it manually 😊?

mjimenez98 replied:
_Manually_ 🥴

I'm sorry 😭  In [this excerpt](https://github.com/mjimenez98/message-board/pull/27/files#diff-9f7468293ceb1b36c8984e8cc2f57426f7556a4f554810d2767f595764859149R18-R24) for example the closing tag of the if is not aligned with the declaration.

But it's been approved so please don't feel like you have to do this. The front-end will be revamped soon.

CoBatool commented:
> This is pretty cool! Love it ❤️
> 
> One non-blocking thing I noticed is that if I click on edit and I don't change anything, the message will still pop up saying I edited the post. Would it be possible to verify if the original title or message has changed before displaying the edit timestamp?
> 
> Also, I would suggest running `Code -> Reformat Code` or `Alt + Command + L` on `PostServlet.java` and `DBPost.java` to get the formatting in line with main



> This is pretty cool! Love it ❤️
> 
> One non-blocking thing I noticed is that if I click on edit and I don't change anything, the message will still pop up saying I edited the post. Would it be possible to verify if the original title or message has changed before displaying the edit timestamp?
> 
> Also, I would suggest running `Code -> Reformat Code` or `Alt + Command + L` on `PostServlet.java` and `DBPost.java` to get the formatting in line with main

Great!!! Thanks I changed it so the timestamp would be shown only if it is different than the original post.  Also fixed the formatting, thank you ! 🧐

mjimenez98 commented:
@CoBatool It is trivial but I just realized a potential experience issue: if the contents of the message are too long, it will be pretty hard to edit it since you will need to scroll right to fix, for example, a typo.

Would it work if the message input was changed to a textarea?

CoBatool commented:
> @CoBatool It is trivial but I just realized a potential experience issue: if the contents of the message are too long, it will be pretty hard to edit it since you will need to scroll right to fix, for example, a typo.
> 
> Would it work if the message input was changed to a textarea?

AH, good point!!! Fixed ✨
